google-cloud-spanner/src/main/java/com/google/cloud/spanner/InstanceConfigRpcCache.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import com.google.cloud.spanner.spi.v1.GapicSpannerRpc;
+import com.google.cloud.spanner.spi.v1.SpannerRpc;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.protobuf.FieldMask;
+import com.google.spanner.admin.instance.v1.GetInstanceRequest;
+import com.google.spanner.admin.instance.v1.Instance;
+import com.google.spanner.admin.instance.v1.InstanceName;
+import com.google.spanner.v1.DatabaseName;
+import com.google.spanner.v1.SessionName;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class InstanceConfigRpcCache {
```

Comment:
Would it be possible to move this class to the `com.googlel.cloud.spanner.spi.v1` package and make it package-private? It is not a class that any users should have to deal with, so keeping it package-private whenever possible would be preferable. This also reduces the chances that we run into unnecessary binary compatibility check failures in the future.

Reply:
Done